packages/jest-core/src/SearchSource.ts

```diff
@@ -272,9 +272,15 @@ export default class SearchSource {
       const options = {nocase: true, windows: false};
 
       paths = paths
-        .map(p => path.resolve(this._context.config.cwd, p))
-        .map(p => micromatch(allFiles, p.replace(/\\/g, '\\\\'), options)[0])
-        .filter(p => p);
+        .map(
+          p =>
```

Comment:
Could we extract to variables for some ease of reading?

```js
.map((path) => {
  const relative = path
    .resolve(this._context.config.cwd, p)
    .replace(/\\/g, '\\\\');

  const matches = micromatch(
    allFiles,
    path.resolve(this._context.config.cwd, p).replace(/\\/g, '\\\\'),
    options,
  );

  return matches[0];
});
```

or something like that 🙂